source/common/quic/envoy_quic_client_stream.cc

```diff
@@ -44,8 +44,16 @@ Http::Status EnvoyQuicClientStream::encodeHeaders(const Http::RequestHeaderMap&
 
   local_end_stream_ = end_stream;
   SendBufferMonitor::ScopedWatermarkBufferUpdater updater(this, this);
-  auto spdy_headers = envoyHeadersToHttp2HeaderBlock(headers);
-  if (headers.Method()) {
+  spdy::Http2HeaderBlock spdy_headers;
+  if (Http::Utility::isUpgrade(headers)) {
+    // transform Upgrade from H1 to H3
```

Comment:
This comment is somewhat redundant with the function "transformUpgradeRequestFromH1toH3". Instead it might be a good idea to explain *why* this transformation is happening. Perhaps something like:

"Upgrade requests are represented in their HTTP/1 form, regardless of the protocol used to send them, so they must be converted to their HTTP/3 form before sending them to the peer"

https://google.github.io/styleguide/cppguide.html#Implementation_Comment_Donts